### PR TITLE
Add polaris validations GHA

### PR DESCRIPTION
**Pull request by Jaanki**
[Polaris](https://polaris.docs.fairwinds.com/) is a k8s validation tool. It runs a variety of checks to ensure
that Kubernetes pods and controllers are configured using best
practices, helping you avoid problems in the future.

https://github.com/submariner-io/submariner-operator/pull/1860 is a result of red flag raised by Polaris.

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: .github/workflows/polaris.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        run: |
+          kubectl --kubeconfig output/kubeconfigs/kind-config-cluster1 label node cluster1-worker submariner.io/gateway=true --overwrite
+          kubectl --kubeconfig output/kubeconfigs/kind-config-cluster2 label node cluster2-worker submariner.io/gateway=true --overwrite
+
+      - name: Join clusters
+        run: |
+          bin/subctl join --kubeconfig output/kubeconfigs/kind-config-cluster1 broker-info.subm --clusterid cluster1 --natt=false \
+            --servicecidr 100.1.0.0/16 --clustercidr 10.1.0.0/16
+          bin/subctl join --kubeconfig output/kubeconfigs/kind-config-cluster2 broker-info.subm --clusterid cluster2 --natt=false \
+            --servicecidr 100.2.0.0/16 --clustercidr 10.2.0.0/16
```
Comment (skitt) [inline .github/workflows/polaris.yml:44]: `make deploy` should be able to replace all this.

**Review by skitt — CHANGES_REQUESTED** [review]

File: .github/workflows/polaris.yml (hunk 1 above)
Comment (Jaanki) [reply .github/workflows/polaris.yml:44]: I think deploy does e2e/other tests as well.

Comment (skitt) [reply .github/workflows/polaris.yml:44]: It doesn’t run `e2e` (that’s `make e2e`); it brings up Submariner on two clusters, waits for connectivity to be established, and runs a basic connectivity test.

**Comment by dfarrell07** [comment]
Discussing this on PR sync, closing for now can re-open if there's interest.

**Pull request closed without merging**